pkg/client/kubernetes/clientmap/internal/shoot_clientmap.go

```diff
@@ -120,6 +121,34 @@ func (f *ShootClientSetFactory) NewClientSet(ctx context.Context, k clientmap.Cl
 	return clientSet, err
 }
 
+func (f *ShootClientSetFactory) getSeedNamespace(ctx context.Context, key ShootClientSetKey) (string, kubernetes.Interface, error) {
+	gardenClient, err := f.GetGardenClient(ctx)
+	if err != nil {
+		return "", nil, fmt.Errorf("failed to get garden client: %w", err)
+	}
+
+	shoot := &gardencorev1beta1.Shoot{}
+	if err := gardenClient.Client().Get(ctx, client.ObjectKey{Namespace: key.Namespace, Name: key.Name}, shoot); err != nil {
+		return "", nil, fmt.Errorf("failed to get Shoot object %q: %w", key.Key(), err)
+	}
+
+	if shoot.Spec.SeedName == nil {
+		return "", nil, fmt.Errorf("shoot %q is not scheduled yet", key.Key())
+	}
+
+	seedClient, err := f.GetSeedClient(ctx, *shoot.Spec.SeedName)
+	if err != nil {
+		return "", nil, fmt.Errorf("failed to get seed client: %w", err)
+	}
+
+	project, err := ProjectForNamespaceWithClient(ctx, gardenClient.Client(), shoot.Namespace)
+	if err != nil {
+		return "", nil, fmt.Errorf("failed to get Project for Shoot %q: %w", key.Key(), err)
+	}
+
+	return shootpkg.ComputeTechnicalID(project.Name, shoot), seedClient, nil
```

Comment:
The technicalID should be also present in the Shoot status. Would it be possible to get it from there instead of computing it again?

Reply:
`ComputeTechnicalID` takes a look at the `technicalID` in the shoot status and falls back to the computation if it is not set yet.
This is just how it was done before in `InitializeShootClients`, I just moved the code.
DYT we should only rely on the status?

Reply:
You are right, I  missed that part in `ComputeTechnicalID`. Probably we can skip the `ProjectForNamespaceWithClient` invocation if the shoot already has the technicalID set.

Reply:
Well, good point. Done 👍